Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/TestCertificateRepository.kt

```diff
@@ -129,6 +130,17 @@ class TestCertificateRepository @Inject constructor(
             scope = appScope
         )
 
+    fun findCertificateDetails(containerId: TestCertificateContainerId): Flow<TestCertificate?> =
+        internalData.data.map { map ->
+            val container = map[containerId]
+            if (container?.isNotRecycled == true) {
+                container.toTestCertificateWrapper(
+                    valueSetsRepository.latestTestCertificateValueSets.first(),
+                    dccValidityMeasuresObserver.dccValidityMeasures.first()
+                )?.testCertificate
+            } else null
+        }
```

Comment:
I think this can be simplified to 
```kotlin
            map[containerId].takeIf { it?.isNotRecycled == true }?.toVaccinationCertificateWrapper(
                valueSetsRepository.latestVaccinationValueSets.first(),
                dccValidityMeasuresObserver.dccValidityMeasures.first()
            )?.vaccinationCertificate
```

Reply:
and small unit test please ?